src/sentry/tasks/sentry_apps.py

```diff
@@ -303,13 +303,23 @@ def send_and_save_webhook_request(url, sentry_app, app_platform_event):
         resp = safe_urlopen(
             url=url, data=app_platform_event.body, headers=app_platform_event.headers, timeout=5
         )
+        resp.raise_for_status()
 
-    except RequestException:
+    except Timeout as e:
         track_response_code("timeout", slug, event)
         # Response code of 0 represents timeout
         buffer.add_request(response_code=0, org_id=org_id, event=event, url=url)
         # Re-raise the exception because some of these tasks might retry on the exception
         raise
+
+    except HTTPError as e:
+        status_code = e.response.status_code
+        track_response_code(status_code, slug, event)
+        # Use the response code from the error
+        buffer.add_request(response_code=status_code, org_id=org_id, event=event, url=url)
```

Comment:
@NisanthanNanthakumar We need the `error_id` and `project_id` when calling `add_request` so we can link the error for the Sentry dashboard